### PR TITLE
NAS-132961 / 25.04 / Move enclosure label to its own file

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 1:00

File: src/middlewared/middlewared/alembic/versions/25.04/2024-12-16_12-49_enclosure_label.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# revision identifiers, used by Alembic.
+revision = '19cdc9f2d2df'
+down_revision = 'b44c092bfa30'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.execute("ALTER TABLE truenas_enclosurelabel RENAME TO enclosure_label")
```
Comment (yocalebo) [inline src/middlewared/middlewared/alembic/versions/25.04/2024-12-16_12-49_enclosure_label.py:21]: The table will have entries with `encid`'s that don't match with what `enclosure2.query` returns so how does a user remove those entires?

File: src/middlewared/middlewared/plugins/enclosure_/enclosure_label.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                "datastore.delete", "enclosure.label", [["encid", "=", id_]]
+            )
+            await self.middleware.call(
+                "datastore.insert",
+                "enclosure.label",
+                {"encid": id_, "label": data["label"]},
+            )
+
+    @private
+    async def get_labels(self):
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/enclosure_/enclosure_label.py:34]: Why can we not do `enclosure.label.query`?

File: src/middlewared/middlewared/alembic/versions/25.04/2024-12-16_12-49_enclosure_label.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/alembic/versions/25.04/2024-12-16_12-49_enclosure_label.py:21]: We can automatically remove the labels for non-existing IDs on middleware startup. Will this be safe?

**Comment by themylogin** [comment]
> This is still naming the plugin namespace `enclosure`. Please rename the class to `EnclosureLabelService` and explicitly define `namespace = "enclosure.label"`

So, API change? Can we simplify the API then?

**Comment by yocalebo** [comment]
> > This is still naming the plugin namespace `enclosure`. Please rename the class to `EnclosureLabelService` and explicitly define `namespace = "enclosure.label"`
> 
> So, API change? Can we simplify the API then?

Yes absolutely, let's simplify as much as possible.

**Comment by themylogin** [comment]
WebUI PR https://github.com/truenas/webui/pull/11200

**Review by yocalebo — APPROVED** [review]

**Comment by themylogin** [comment]
time 1:30

**Pull request merged**